### PR TITLE
Enhance ph-debug

### LINKED ISSUE #538: Enhance ph-debug
I really just wanted to be able to start and stop links from ph-debug, but once I got in there, there were a bunch of changes to make

### PR DESCRIPTION
**Pull request by svjaffer**
Did a bunch of cleanup in Ph-debug.  Main changes are:
- CLI mode
- Switched over to using the clap Subcommands, which cleaned some things up
- Added an RpcCommands enum to keep commands consistent between PH and PH-debug

resolves #538

### CODE REVIEW

File: adapter/ph-debug/src/main.rs [hunk 1]
```diff
@@ -17,92 +18,236 @@ use std::os::unix::net::UnixStream;
...
+use zpr::rpc_commands::RpcCommands;
 use zpr_ext::std::os::unix::net::{SocketAncillary, UnixStreamExt};
 const ANCILLARY_BUFFER_SIZE: usize = 128;
 
+macro_rules! basic_command {
+    ($comm:expr, $socket:ident) => {
+        basic_call_response_0($comm, $socket)
+    };
+    ($comm:expr, $socket:ident, $arg:tt) => {
+        basic_call_response_1(($comm), ($socket), ($arg))
```
Comment (cpacejo) [inline adapter/ph-debug/src/main.rs]: Not an error but I think the parentheses should be unnecessary here?  (as with the other two cases)

File: adapter/ph-debug/src/main.rs [hunk 2]
```diff
@@ -17,92 +18,236 @@ use std::os::unix::net::UnixStream;
...
 
-    #[arg(
-        long,
-        help = "",
-        long_help = "This option has no default, if no program is provided for a command that \
-                            needs a program, no program will be set"
-    )]
-    program: Option<String>,
+    /// Path to the Packet Handler's management socket
+    socket: String,
```
Comment (cpacejo) [inline adapter/ph-debug/src/main.rs]: Possible to keep this as a flag (i.e. `#[arg(short, long)]`)?  I'm imagining in the near-ish future we can then make the flag optional and fall back to a default `/var/run/zpr/ph.sock` or pull it from env vars or such.

**Review by cpacejo — DISMISSED** [review]
Nice!!!!!!!!  I am excited to use the prompt mode!!!

**Review by mkolehmainen — DISMISSED** [review]
looks great!

File: adapter/ph-debug/src/main.rs (hunk 2 above)
Comment (svjaffer) [reply adapter/ph-debug/src/main.rs]: OK, switched it to that default

**Review by mkolehmainen — APPROVED** [review]
looks good to me!

**Pull request merged**